### ISSUE TITLE
RAT proof feature: Countdown uses registration time instead of time of test result as starting point

### ISSUE THREAD

**Issue by Ein-Tim** [corona-warn-app/cwa-documentation#661]
## Avoid duplicates
* [X] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [X] Bug is specific for Android only, for general issues / questions that apply to iOS and Android please raise them in the [documentation repository](https://github.com/corona-warn-app/cwa-documentation)
* [X] Bug is not already reported in another issue

## Technical details

- Device name: Samsung Galaxy S10+
- Android version: 11
- App version: 2.3.4

## Describe the bug
If a user takes a RAT on the 01.06. at 12h but only registers the test result in the Corona-Warn-App on the 02.06. at 12h, [the counter](https://www.coronawarn.app/assets/screenshots/2-4/de/android/rat_7.png), which is shown under Android when a RAT returns negative, does not count the hours since the test result is available (and correctly shows that the test result is available since 24h in the example) but counts the time since the user has registered the test in the app.

## Steps to reproduce the issue

1. Take a RAT at a test site supporting the CWA ([this](https://map.schnelltestportal.de) map helps to find one)
2. Register the test in the CWA one day after the swap was taken and the test result was avaliable
3. See that the counter is wrong, as described above. 

## Expected behaviour

The counter should count the hh:mm:ss since the test result is available, but not the hh:mm:ss since the test result was registered in the app.

## Additional context

This bug report is based on the information a user shared on Twitter: https://twitter.com/Fridde82/status/1408325023429976068?s=20

--- 

## Related Issue
- https://github.com/corona-warn-app/cwa-wishlist/issues/543 

----
Internal Tracking ID: [EXPOSUREAPP-8122](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-8122)
Internal Tracking ID: [EXPOSUREAPP-8156](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-8156)

**Comment by dsarkar** [comment]
@vaubaehn @Ein-Tim Indeed, there is not much that can be done from CWA side. This issue has been raised and discussed with TSI. That is all I know at this stage.

**Comment by Ein-Tim** [comment]
@dsarkar 

Thank you for the update!
I think it would make sense to deactivate the proof feature if the test center didn't transmit a AC timestamp. 
I'll open a new wishlist item for this later on.

**Comment by vaubaehn** [comment]
Hi @dsarkar , do you have information what the fix im 2.10 will encompass?

**Comment by dsarkar** [comment]
@vaubaehn The internal fix 2.10 was removed. As far as I understand, the corresponding RAT partners are contacted and advised to set adequately the  "sc" timestamp.

**Comment by vaubaehn** [comment]
@dsarkar @heinezen 
> @vaubaehn The internal fix 2.10 was removed. As far as I understand, the corresponding RAT partners are contacted and advised to set adequately the "sc" timestamp.

So, if I understand it correctly, "corresponding RAT partners" are only the ones with a bad "sc" timestamp? This would mean, in case we (as in: "the CWA users") find a RAT partner with a bad "sc" timestamp by chance in the future, we will report it to you, and you will notify them?
If this is the case, nothing changed, and everything stayed as before:
- TSI will continue to not detect any "sc" timestamp problem during the "Abnahmetest" (CWA quicktest and DCC), to prevent RAT partners going online with a bad "sc" timestamp implementation
- TSI will not notify **_all_** RAT partners via info mailing, that RAT partners should check their "sc" timestamp of correct implementation (that "sc" only covers the time when the sample was collected, and nothing else), like "Aus aktuellem Anlass möchten wir unsere Partner darauf hinweisen, dass... und bitten Sie zu überprüfen, ob..." and providing information on how the "sc" timestamp should be implemented
- TSI won't adapt/correct their documentation on "cwa-quicktest-onboarding" to give better and clearer information for RAT partners on "sc" timestamp?
- TSI won't change the behaviour for cwa-testresult-server/cwa-quicktest-backend, to only accept POSTs for testresults with "sc" timestamp provided? (Currently: optional. When no "sc" timestamp provided when submitting test results, the system adds the corresponding timestamp)
- It is still possible, that there are RAT partners out there, which have a bad (or NO!) implementation for their "sc" timestamp, and when CWA users are able to find them, they can continue to abuse the system and are able to manipulate the validity period for a RAT?

If this is the case, then actually nothing changes.

Maybe @daniel-eder can please make an exception to former behaviour (resting in silence) and please provide more information on this issue?

**Comment by daniel-eder** [comment]
I do not have any updated information about the timestamp handling, but I brought up this topic with the colleagues in development to get some eyes on this issue.

**Comment by ascheibal** [comment]
Hi @vaubaehn,
I will comment on the technical issues you mentioned.
- The onboarding procedure is described under https://github.com/corona-warn-app/cwa-quicktest-onboarding/wiki/ there is also a section for "Abnahmetest" focus on the topics you mentioned, the procedure would determine a wrong implementation of sc feature. As you can see there information is up-to-date and will be changed if needed.
- "...behaviour for cwa-testresult-server/cwa-quicktest-backend...". All RAT testresults on testresult server are delivered via a partner interface, that the partners have managed/controlled access to. This interface accepts the testId (mandatory), testresult (mandatory) and the sample collection timestamp, sc (optional). As this interface developed over time, 'sc' was not part of the initial release, came for kind-of 'validity duration' feature in cwa, now superceded by the DCC. As not all of the partners where able to deliver the sc right in time, the first ones were willing to start, don't want to be blocked by others. That is why it is optional. 
To be clear: In case 'sc' is not provided, it is set by the testresult server **to the time when the test result arrives** from the partner at the interface. This is **not** the time the user registers the tests/scanns the QRCode. As result the cwa mobile app **always** gets a timestamp on their interface at verification server, which in fact fulfills your requirement of making sc mandatory.  When a user scans a test 24h after it was delivered to testresult server, the app will show that with an elapsed time counter showing the difference 'now-sc', which then will 24h.
As we further can asume, that a partner delivers the test to CWA backend immedeatly after the result is fixed, the time difference between the real time of 'sample collection' and the automatic sc timestamp in testresult server is around 20 minutes, when the partner does not provide the sc, which is close enough for the purpose it is intended (->Show a non-dcc test result with time elapsed since test).
Hope I could mostly destroy you apprehensions.

**Comment by vaubaehn** [comment]
Hi @daniel-eder and @ascheibal ,

it looks like you are not willingful to follow/examine this subject further.

I am still convinced that as long as the infrastructure (flow between 3rd PoC parties and affected CWA backends) open opportunities for users to abuse the system by manipulating the validity period of their test DCCs, we are talking about a **security breach** here, whereas "security" refers to public health/health of individuals.

So I now consider to hand this subject over to other experts. Maybe Heise Security could be a good option - or do you have better recommendations?

Cheers, v.

**Comment by daniel-eder** [comment]
@vaubaehn I brought this up again with @ascheibal to review your message in depth again and check against our understanding of this. (As a side note, as you may have noticed, for our development-involved colleagues it is hard to keep track of GitHub notifications, they are sometimes in the tens or hundreds daily over the different covid-related projects, hence at times an internal ping is needed on top of things to schedule such a review) 

That said, if you know others that are willing to explain their own understanding of this (or other) potential security issues where you think our security team are not aware of the full scope of the threat, please do so!

**Comment by vaubaehn** [comment]
@daniel-eder 
Lieber Herr Eder, 
vielen Dank für die Antwort. Es ist bestimmt gut, wenn sich @ascheibal das Problem nochmal genauer anschaut. Ich werde in einem weiteren Kommentar auf deutsch erneut versuchen, dass Problem besser und ohne sprachliche Verzerrungen (da Englisch ja nicht unsere Muttersprache ist), auf den Punkt zu bringen. Das wird jetzt einen Augenblick dauern, aber ich melde mich nachher wieder.
Viele Grüße,
v.

**Comment by vaubaehn** [comment]
@ascheibal @daniel-eder 
Hier die angekündigte deutsche Version:

**Testlinge können den Gültigkeitszeitraum ihres Schnelltests ("native" Anzeige in der CWA und ggf. DCC) manipulieren**, wenn folgende Voraussetzungen erfüllt sind:
- Der Testanbieter hat eine eigene Softwarelösung implementiert, und übermittelt seine Testergebnisse direkt an das CWA-backend. Ob auch das CWA-quick-test-frontend von der Sicherheitslücke betroffen sein kann, müsste noch geklärt werden (habe ich bislang nicht weiter verfolgt).
- Die Testlinge buchen einen Schnelltesttermin online und hinterlassen ihre E-Mail-Adresse, über die sämtliche Nutzerkommunikation (automatisiert) erfolgt. Die Testlinge lassen zum vereinbarten Termin den Schnelltest durchführen.
- Der Testanbieter schickt einen Bestätigungslink per E-Mail an die Getesteten, über den die Testdurchführung quittiert werden muss (Hinweis: seit dem 01.07.2021 **_muss_** die Testdurchführung für einen kostenlosen (Bürger-)Test quittiert werden, da der Testanbieter ansonsten die Kosten für die Durchführung nicht erstattet bekommt).
- Der Testanbieter übermittelt das Testergebnis an das quicktest-backend **_erst nachdem_** die Getesteten die Testdurchführung quitttiert haben.
- Der Testanbieter übermittelt dabei **_keinen `sc `timestamp_** (`sc `= sample collected; Zeitpunkt, wann die Probe vom Testling genommen wurde). Zum jetzigen Zeitpunkt ist die Übermittlung eines `sc `timestamps für die Testnabieter immer noch **_optional_**!
- Für das DCC: Der Testanbieter setzt fälschlicherweise einen falschen Zeitpunkt für `sc`.

Wenn diese Voraussetzungen zusammen fallen, können Getestete so einfach die in der CWA angezeigte Gültigkeit für den "nativen" Schnelltest (aus der Zeit vor dem DCC) manipulieren:
- Testling geht zum Schnelltest.
- Getestete erhalten per E-Mail den Bestätigungslink für die Testdurchführung, **_aber bestätigen die Testdurchführung noch nicht_**!
- Getestete warten bis zu dem Zeitpunkt, ab wann der Test gültig sein soll (z. B. bis unmittelbar vor einem Club-Besuch), und bestätigen dann die Testdurchführung per Klick auf den Link.
- Das Backend des Testanbieters sendet nun das Testergebnis ohne `sc` timestamp an das CWA-quicktest-backend/an den CWA-testresult-server. Da der `sc` timestamp fehlt, **_setzt nun der CWA-testresult-server den `sc `timestamp automatisch auf den Zeitpunkt des Eingangs des Testergebnisses_**!
- Die CWA holt sich das Testergebnis vom testresult-server, **_aber der Schnelltest hat nun als Zeitpunkt der Testdurchführung nicht den Zeitpunkt der Probenentnahme, sondern den Zeitpunkt der Bestätigung der Testdurchführung!_**

**_Ist dies ein realistisches Szenario? Ja, denn genau das war bereits ein Problem bei einem der größten Testanbieter in Deutschland!_**
Und zwar bei NETSYNO, die unter anderem die Testabwicklung für die Drogeriekette _dm_ übernommen hatten. **Es ging hier also um hunderttausende von Tests.** Wie kam es dazu? Als NETSYNO an den Markt ging, mussten Bürgertests noch nicht quittiert werden. Da die Übermittlung eines `sc` timestamps optional war/ist, verzichtete NETSYNO darauf - denn, genau wie @ascheibal oben auch argumentierte, wenn das Testergebnis zeitnah an das CWA-backend gesendet wird, entspricht der Übermittlungszeitpunkt ungefähr dem `sc `Zeitpunkt. Nachdem aber die Notwendigkeit der Quittierung der Testdurchführung zum 01.07. eingeführt wurde, baute NETSYNO sein Backend entsprechend um: es wurde ein Bestätigungslink für die Testdurchführung versendet. Aber: der `sc `timestamp wurde zunächst auch weiterhin nicht übermittelt, was dazu führte, dass für die CWA der Zeitpunkt der Probenentnahme nun der Zeitpunkt war, den der testresult-server automatisch gesetzt hatte. **_Es lag nun also in der Hand des Testlings (durch die Bestätigung der Testdurchführung) darüber zu bestimmen, welcher Zeitpunkt der Testdurchführung in der CWA angezeigt wurde!_**

Nachdem **bereits am 27. Juni** @Ein-Tim darauf aufmerksam machte, dass mit dem angezeigtem `sc `timestamp [in der CWA etwas nicht stimmt](https://github.com/corona-warn-app/cwa-documentation/issues/661#issue-948590248), meldete dann @RaoulDuke25 am 04. Juli ein [ähnliches Problem für den Testanbieter testflow.eu](https://github.com/corona-warn-app/cwa-documentation/issues/661#issuecomment-873569224). Das nahm ich dann zum Anlass, das Problem um den 09.07. herum gezielt zu testen, und dafür musste dann leider dm (=NETSYNO) herhalten, da bei mir um die Ecke. Und tatsächlich: mit dem immer noch bestehenden Design ließ sich [diese Sicherheitslücke auch für diesen Anbieter belegen](https://github.com/corona-warn-app/cwa-documentation/issues/661#issuecomment-877199316). 
Ich meldete das Problem darauf unmittelbar bei NETSYNO, und nicht einmal einen halben Tag später war es hier gefixt: Einfach dadurch, dass nun auch ein (korrekter) `sc` timestamp zusammen mit dem Testergebnis an das CWA backend übermittelt wurde. **Die Reaktion von NETSYNO war hier vorbildlich.** Zudem bin ich der Ansicht, dass NETSYNO keinerlei Vorhaltungen gemacht werden können, da sie sich einfach nur an die von TSI (immer noch) veröffentlicheten Implementierungs-Guidelines gehalten hatten.

Natürlich meldete ich das Problem auch hier, hatte dazu zunächst @ascheibal und @mschulte-tsi angepingt, nach Tagen bis Wochen auch noch weitere TSI-Leute - aber bis heute leider ohne wirklich _angemessene_ Reaktion. Auch SAP hatte (den Rückmeldungen hier nach) anfangs versucht, TSI für das Problem zu sensibilisieren.

**_Dabei ist bis jetzt immer noch nicht klar, wie viele Testanbieter dieses Problem betrifft!_**
Dies kann zuverlässsig nur darüber bestimmt werden, wenn mitgeloggt wird, wie viele Tests/von welchen Anbietern Testergebnisse ohne `sc `timestamp übermittelt werden.

_**Welche Testanbieter sind nicht betroffen?**_
Alle Testanbieter, die sich die Bestätigung für die Testdurchführung nicht online einholen, oder die das Testergebnis unabhängig von der Bestätigung zeitnah nach der Probenentnahme an das CWA-Backend übermitteln, oder die das Testergebnis (ggf. nach der Bestätigung) mit einem _**nachweislich korrekten**_ `sc` timestamp übermitteln. 

**_Aber das geschilderte Problem betrifft anscheinend doch nur die "native"/ursprüngliche Anzeige des Schnelltestergebnisses in der CWA - ist die nicht längst durch das DCC überholt? NEIN!_**
Ich erzähle Ihnen bestimmt nichts Neues, dass es sehr viele kleine Testanbieter, vor allem in den ländlichen Regionen, gibt, die zwar an die CWA angebunden sind (seit 01.08. Pflicht), aber keine DCCs ausstellen (da die zusätzliche Implementierung gerade für kleinere Anbieter zu teuer sein kann, wenn externes Know-How eingekauft werden muss). Es besteht also berechtigter Grund zu der Annahme, dass in Regionen, in denen DCCs kaum verbreitet sind, die "nativen" CWA-Schnelltestergebnisse von Restaurants, Clubs, und anderen Einrichtungen immer noch akzeptiert werden. Oder umgekehrt: es gab bis jetzt noch kein offizielles Verbot der Anerkennung des nativen CWA-Schnelltestergebnisses. **Daher sollte dieses Problem immer noch dringend behoben werden!**

_**Und DCCs sind nicht betroffen? Das kommt darauf an!**_
Ob DCCs auch betroffen sind, kommt darauf an, ob der Testanbieter alle timestamps für das DCC korrekt setzt/implementiert hat. Und hier darf begründet gezweifelt werden, ob das bei _allen_ Anbietern _immer_ der Fall ist, denn wir hatten hier schon verschiedene Meldungen, dass das nicht der Fall war (z. B. mehrfache Meldungen für Probleme bei EcoCare: #689, #668). Wenn z.B. ein Testanbieter für das DCC fälschlicherweise den Zeitstempel für die Bestätigung der Testdurchführung heranzieht (klicken auf den Confirmation-Link) statt des Zeitpunkts der Probenentnahme, hätten wir das selbe Problem auch bei den DCCs.
**Das Problem ist: ist der Testanbieter im Status PROD, lässt sich die Richtigkeit des `sc `timestamps in DCCs von TSI-Seite nicht mehr kontrollieren - da alle Daten verschlüsselt übermittelt werden!** Der einzige Ansatzpunkt, um heraus zu bekommen, ob ein Testanbieter alle Zeitstempel für das DCC richtig setzt, ist der Abnahmetest, bevor der Anbieter online geht.

**_Aber der Abnahmetest stellt doch sicher, dass im "nativen" CWA-Schnelltestergebnis/DCC alle Zeitstempel korrekt gesetzt sind? IN DEM JETZIGEN VERFAHREN HÖCHSTWAHRSCHEINLICH NICHT!_**
Für das native CWA-Schnelltestergebnis gilt:
Der Abnahmetest erfolgt als Videokonferenz, die für einen Zeitraum von 30 bis 60 Minuten angesetzt ist. In dieser Zeit werden ggf. mehrere Testungsszenarien durchgespielt. **Das Problem hierbei: die Abfolge der einzelnen Schritte erfolgt so rasch hintereinander, dass es höchstwahrscheinlich nicht möglich ist, zu bestimmen, welche Aktion den `sc `timestamp auslöst/setzt.**  War es ein manuell eingegebener Zeitpunkt einer fiktiven Probenentnahme (korrekt)? Oder kommt der `sc `timestamp durch eine andere vom Testanbieter ausgelöste Aktion zustande (inkorrekt)? Oder setzt der Anbieter gar keinen `sc `timestamp und das Ergebnis wird durch den testresult-server gesetzt (**Sicherheitslücke**)?
Für das RAT DCC gilt:
Offenbar werden DCCs gar nicht von TSI überprüft! Es obliegt allein dem Testanbieter, ein korrektes und gültiges DCC zu erstellen.
**Obwohl das RKI offiziell Aussteller der DCCs ist und ein korrektes Testergebnis zertifiziert, wird von TSI nicht überprüft, ob die RAT DCCs auch tatsächlich korrekt und gültig sind!**

**_Wie kann das Problem beseitigt werden?_**
**Bereits am 12. Juli** hatte ich notwendige Maßnahmen hier skizziert: https://github.com/corona-warn-app/cwa-quicktest-onboarding/issues/24#issuecomment-878084791 . Die technischen Änderungen sind trivial, die praktischen Umsetzungen leider mit nicht unerheblichem Aufwand verbunden:
Technisch:
- Der CWA-testresult-server darf keinen `sc` timestamp mehr setzen!
- Alle Testergebnisse, die von Testanbietern via API-Aufruf an das CWA-backend übermittelt werden, müssen vom Server abgelehnt werden, wenn kein `sc `timestamp enthalten ist.
- Auch das quick-test-frontend sollte eine manuelle Eingabe des Zeitpunkts der Probenentnahme unterstützen, falls das noch nicht der Fall ist, und das cwa-quick-test-backend nur Übermittlungen mit `sc `timestamp akzeptieren.
- TSI muss eine Möglichkeit schaffen, beim Abnahmetest für die Testanbieter den Zeitpunkt der fiktiven Probenentnahme _exakt (minutengenau)_ mit dem `sc `timestamp abgleichen zu können.
- Die Dokumentation für die Testanbieter muss entsprechend geändert und veröffentlicht werden: "`sc` timestamp nicht mehr optional, sondern _obligat_! Für DCCs ist ein korrekter `sc` timestamp ebenfalls essenziell!"

Praktisch:
- Die technischen Änderungen müssen an die Testanbieter kommuniziert werden: Nach einer Übergangszeit (von ca. 2-3 Wochen?) müssen `sc` timestamps übermittelt werden, ansonsten wird das Testergebnis vom Server nicht mehr akzeptiert - dafür sind ggf. Anpassungen an der Software bei den Testanbietern notwendig. Der `sc `timestamp _muss_ dem Zeitpunkt der Probenentnahme entsprechen. Es ist sinnvoll, eine Erinnerungsmail ca. 1 Woche vor Ablauf des Übergangszeitraums zu senden.
- Bei den Abnahmetests für neue Testanbieter _muss_ überprüft werden, ob der `sc `timestamp korrekt für das native CWA-Schnelltestergebnis gesetzt wird. Wenn der Testanbieter DCCs implementiert hat, _muss_ der Inhalt des DCCs auf Korrektheit/Gültigkeit überprüft werden.
- Es wäre dringend angeraten, auch bei den bereits bestehenden Testanbietern eine Nachprüfung zu den `sc `timestamps und DCC-Inhalten durchzuführen, **sofern nicht bereits _eindeutig_ bekannt ist**, dass die timestamps für die CWA/DCC korrekt sind.

**_Und wenn das Problem nicht beseitigt wird?_**
Dann besteht zumindest die Möglichkeit, dass es da draußen immer noch Testanbieter gibt, bei denen das Zusammenspiel zwischen eigener Software-Implementierung und CWA-backend dazu führt, dass Testlinge den Gültigkeitszeitraum ihres Schnelltests (via nativer Anzeige in der CWA oder per DCC) manipulieren können. **Das kann fatale Konsequenzen haben, sowohl gesundheitliche als auch ökonomische (bei Quarantäne)**, wenn z. B. für einen 3G-Club-Besuch ein maximal 6 Stunden alter Schnelltest erforderlich ist (wie in Schleswig-Holstein), und ein findiger Testling heraus bekommt, dass er sein Schnelltest-Ergebnis manipulieren kann. Er lässt sich z. B. einen Tag vor dem Club-Besuch negativ testen, aber setzt den Testzeitpunkt auf 3 Stunden vor dem Club-Besuch durch den Klick auf den Bestätigungslink des Testanbieters. Wenn aber in den 24+ Stunden nach seiner Testung seine (asymptomatische?) COVID-Erkrankung ausbricht, kann er schön alle Anwesenden im Club anstecken. Gleiches gilt natürlich für fehlerhaft ausgestellte DCCs.

---

Wenn ich verantwortlich bei TSI wäre. würde ich vermutlich als erstes das RKI benachrichtigen, die Problemlage schildern und mögliche Optionen aufzeigen, und mit entsprechnender Rückendeckung der verantwortlichen Institution entsprechend weiter agieren.

Ich würde mich freuen, wenn wir bis Freitagabend von Ihnen eine Rückmeldung über eine (Neu-)Einschätzung des Problems an dieser Stelle erhalten könnten. Ansonsten würde ich es vorziehen, die Öffentlichkeit über die Medien einzubeziehen.

Herzlichen Dank vorab, viele Grüße
v.

**Comment by daniel-eder** [comment]
@vaubaehn Danke für die ausführliche Erklärung auch auf Deutsch - Ich habe das Security Team, das diesen Prozess ursprünglich reviewed hat, spezifisch auf diesen letzten Kommentar hingewiesen.

**Comment by Ein-Tim** [comment]
Is there any change planned here?

**Comment by Ein-Tim** [comment]
@larswmh 

As this is the issue tracker for the Corona-Warn-App developed by both, SAP & T-Systems, I don't think issues should be closed only because SAP doesn't feel responsible, but T-Systems is. Please reopen this issue until the mitigations are finished.

**Comment by svengabr** [comment]
@Ein-Tim the community managers don't have access to TSI Jira, the latest comment related to TSI was:

>to be clear, the sc flag provided by the partner as part of the testresult is not connected to the parameter 6 in the HCert Container of the DCC and is for the use case, that the user does not want a DCC and only see the test in the app.
For the DCC we must rely on the content the Partner sends to the backend, cause it is encrypted and can only be decrypted on mobile app side - hence no validity check (is parameter 6 existing, is parameter 4 existing, are they reasonable concerning current date) can be done by the backend system.
>
>The sc flag is provided to the mobile app as part of the testresult via verification servers /testresult endpoint in every case, so mobile app can rely on it, but it is not signed in any way, so no false certification as you mentioned.
>
>Is your suggestion to make sc key mandatory for the testresult delivered by the partner?
>1.) This would be a breaking change in api, with the need to introduce an additional endpoint due to backward compatibility.
>2.) In the case you mentioned, the sc could then still be 7 days behind deliver date, if partner does not send it to us right in time.
In fact this does not prevent the failure. How should the backend deal with outdated tests, reject them at all?
>
>When we introduced the sc feature, we were urgently asked by some Partners to come up with it, as they had the case, that people are registrate for tests on some day and show up a day earlier e.g. which lead to those negative time counters in the app.
So most partners have special interest to use the sc flag, for those who do not, we have the fall back in place.
to summarize, I see no need to change the code here.

Followed by:
> Change on TSI side is too much effort. If we receive new request we have to inform the test partner directly

So, I think it was correct to close this issue as there is nothing more to be expected to happen for this issue unfortunately.

**Comment by Ein-Tim** [comment]
Thanks @svengabr.


@vaubaehn FYI ⬆️
